pyfr/backends/base/generator.py

```diff
@@ -12,10 +12,10 @@ def __init__(self, name, spec, body):
         self.name = name
 
         specptn = r'''
-            (?:(in|inout|out)\s+)?                           # Intent
-            (?:(broadcast(?:-row|-col)|mpi|scalar|view)\s+)? # Attrs
-            ([A-Za-z_]\w*)                                   # Data type
-            ((?:\[\d+\]){0,2})$                              # Dimensions
+            (?:(in|inout|out)\s+)?                            # Intent
```

Comment:
I'd break the broadcast stuff out into its own commit, as it can go standalone (and @semi-h or @pv101 might want to comment).

Reply:
Giorgio also had to implement this type of broadcasting for the synthetic turbulence generation PR. It can be nice to have this in a separate commit as it extends the DSL's capability.